Sources/Testing/EntryPoints/ABIv0/Encoded/ABIv0.EncodedEvent.swift

```diff
@@ -0,0 +1,103 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension ABIv0 {
+  /// A type implementing the JSON encoding of ``Event`` for the ABI entry point
+  /// and event stream output.
+  ///
+  /// This type is not part of the public interface of the testing library. It
+  /// assists in converting values to JSON; clients that consume this JSON are
+  /// expected to write their own decoders.
+  struct EncodedEvent: Sendable {
+    /// An enumeration describing the various kinds of event.
+    ///
+    /// Note that the set of encodable events is a subset of all events
+    /// generated at runtime by the testing library.
+    ///
+    /// For descriptions of individual cases, see ``Event/Kind``.
+    enum Kind: String, Sendable {
+      case runStarted
+      case testStarted
+      case testCaseStarted
+      case issueRecorded
+      case knownIssueRecorded
+      case testCaseEnded
+      case testEnded
+      case testSkipped
+      case runEnded
+    }
+
+    /// The kind of event.
+    var kind: Kind
+
+    /// The source location of the event, if applicable.
+    var sourceLocation: SourceLocation?
+
+    /// The instant at which the event occurred on the current system's
+    /// suspending clock.
+    var timestamp: Double
+
+    /// The instant at which the event occurred on the wall clock.
+    var timestampSince1970: Double
+
+    /// Human-readable messages associated with this event that can be presented
+    /// to the user.
+    var messages: [EncodedMessage]
+
+    /// The ID of the test associated with this event, if any.
+    var testID: EncodedTest.ID?
+
+    /// The ID of the test case associated with this event, if any.
+    ///
+    /// - Warning: Test cases are not yet part of the JSON schema.
+    var _testCase: EncodedTestCase?
```

Comment:
I know this is non-final indicated by the underscore, and that's good. I think we ought to move in the direction of this being an ID of a test case, rather than encoding the full test case itself inline.

Reply:
We don't encode the entire test case structure here, no.